backfill/src/main.rs

```diff
@@ -252,8 +278,45 @@ fn main() -> eyre::Result<()> {
                 Ok(())
             })
             // Install the backfill ExEx.
-            .install_exex("Backfill", |ctx| async move {
-                Ok(BackfillExEx::new(ctx, exex_backfill_tx, backfill_rx, 10).start())
+            .install_exex("Backfill", move |ctx| {
+                // Rust seems to trigger a bogus higher-ranked lifetime error when using
+                // just an async closure here -- using `spawn_blocking` avoids this
+                // particular issue.
+                //
+                // To avoid the higher ranked lifetime error we use `spawn_blocking`
+                // which will move the closure to another blocking-allowed thread,
+                // then execute.
+                //
+                // Source: https://github.com/vados-cosmonic/wasmCloud/commit/440e8c377f6b02f45eacb02692e4d2fabd53a0ec
+                tokio::task::spawn_blocking(move || {
```

Comment:
I believe this equivalent to ctx.task_executor().spawn_blocking().await

Reply:
yeah, but task executor methods accept futures that resolve to `()`, but we need the return value here